pkg/controller/controller_ref_manager.go

```diff
@@ -516,36 +517,6 @@ func (m *ControllerRevisionControllerRefManager) ReleaseControllerRevision(histo
 	return err
 }
 
-type objectForDeleteOwnerRefStrategicMergePatch struct {
-	Metadata objectMetaForMergePatch `json:"metadata"`
-}
-
-type objectMetaForMergePatch struct {
-	UID              types.UID           `json:"uid"`
-	OwnerReferences  []map[string]string `json:"ownerReferences"`
-	DeleteFinalizers []string            `json:"$deleteFromPrimitiveList/finalizers,omitempty"`
-}
-
-func deleteOwnerRefStrategicMergePatch(dependentUID types.UID, ownerUID types.UID, finalizers ...string) ([]byte, error) {
```

Comment:
One nit: instead of creating that extra package can you move `GenerateDeleteOwnerRefStrategicMergeBytes` here and all the controllers can just use this location, especially that most have it already imported, so it's most likely a no-op for most.